src/providers/provider.ts

```diff
@@ -241,6 +242,7 @@ export abstract class Provider {
 
     abstract async sendTransaction(signedTransaction: SignedTransaction): Promise<FinalExecutionOutcome>;
     abstract async txStatus(txHash: Uint8Array, accountId: string): Promise<FinalExecutionOutcome>;
+    abstract async query(params: object): Promise<any>;
     abstract async query(path: string, data: string): Promise<any>;
```

Comment:
Do we want to have both versions of `query` here?

Reply:
Yes to avoid breaking backwards compatibility